src/native.js

```diff
@@ -280,3 +414,309 @@ function getNativeKeys(adUnit) {
+        }
+      }
+    }
+  }
+  return bidRequests;
+}
+
+export function toOrtbNativeResponse(legacyResponse, ortbRequest) {
+  const ortbResponse = {
+    link: {},
+    assets: [],
+    eventtrackers: []
+  };
+  Object.keys(legacyResponse).filter(key => !!legacyResponse[key]).forEach(key => {
+    const value = legacyResponse[key];
+    switch (key) {
+      // process titles
+      case 'title':
+        const titleAsset = ortbRequest.assets.find(asset => asset.title != null);
+        titleAsset.title = {
+          text: value
+        };
+        ortbResponse.assets.push(titleAsset);
+        break;
+      case 'image':
+      case 'icon':
+        const imageType = key === 'image' ? NATIVE_IMAGE_TYPES.MAIN : NATIVE_IMAGE_TYPES.ICON;
+        const imageAsset = ortbRequest.assets.find(asset => asset.img != null && asset.img.type == imageType);
+        imageAsset.img = {
+          url: value
+        };
+        ortbResponse.assets.push(imageAsset);
+        break;
+      case 'clickUrl':
+        ortbResponse.link.url = value;
+        break;
+      case 'clickTrackers':
+        ortbResponse.link.clicktrackers = value;
+        break;
+      case 'impressionTrackers':
+      case 'javascriptTrackers':
+        ortbResponse.eventtrackers.push({
+          event: 1,
+          method: key == 'impressionTrackers' ? 1 : 2,
+          url: value,
+        });
+        break;
+      default:
+        if (key in PREBID_NATIVE_DATA_KEYS_TO_ORTB) {
+          const dataAsset = ortbRequest.assets.find(asset => asset.data != null && asset.data.type === NATIVE_ASSET_TYPES[PREBID_NATIVE_DATA_KEYS_TO_ORTB[key]]);
+          dataAsset.data = {
+            value
+          };
+          ortbResponse.assets.push(dataAsset);
+        }
+        break;
+    }
+  });
+  return ortbResponse;
+}
+
+/**
+ * Generates a legacy response from an ortb response. Useful during the transition period.
+ * @param {*} ortbResponse a standard ortb response object
+ * @param {*} ortbRequest the ortb request, useful to match ids.
+ * @returns an object containing the response in legacy native format: { title: "this is a title", image: ... }
+ */
+function toLegacyResponse(ortbResponse, ortbRequest) {
+  const legacyResponse = {};
+  const requestAssets = ortbRequest?.assets || [];
+  legacyResponse.clickUrl = ortbResponse.link.url;
+  for (const asset of ortbResponse?.assets || []) {
+    const requestAsset = requestAssets.find(reqAsset => asset.id === reqAsset.id);
+    if (asset.title) {
+      legacyResponse.title = asset.title.text;
+    } else if (asset.img) {
+      legacyResponse[requestAsset.img.type === NATIVE_IMAGE_TYPES.MAIN ? 'image' : 'icon'] = asset.img.url;
+    } else if (asset.data) {
+      legacyResponse[PREBID_NATIVE_DATA_KEYS_TO_ORTB_INVERSE[NATIVE_ASSET_TYPES_INVERSE[requestAsset.data.type]]] = asset.data.value;
+    }
+  }
+  return legacyResponse;
+}
+
+/**
+ * Converts an OpenRTB request to a proprietary Prebid.js format.
```

Comment:
Is the description here accurate for this function?  Per the function name, I would have thought description would have been the other way around.